includes/class-wc-stripe-helper.php

```diff
@@ -373,7 +375,20 @@ public static function get_webhook_url() {
 	public static function get_order_by_source_id( $source_id ) {
 		global $wpdb;
 
-		$order_id = $wpdb->get_var( $wpdb->prepare( "SELECT DISTINCT ID FROM $wpdb->posts as posts LEFT JOIN $wpdb->postmeta as meta ON posts.ID = meta.post_id WHERE meta.meta_value = %s AND meta.meta_key = %s", $source_id, '_stripe_source_id' ) );
+		if ( class_exists( 'OrderUtil' ) && OrderUtil::custom_orders_table_usage_is_enabled() ) {
+			$orders   = wc_get_orders(
+				[
+					'limit'      => 1,
+					'meta_query' => [
+						'key'   => '_stripe_source_id',
+						'value' => $source_id,
+					],
+				]
+			);
+			$order_id = current( $orders )->get_id();
```

Comment:
`current` may return `false` so I think we need to handle that somehow.

```suggestion
			$order_id = current( $orders ) ? current( $orders )->get_id() : false;
```

Returning `false` is really optional. So long as the default value causes `empty()` to evaluate to `true` anything should be fine.